anaDACScan.py

```diff
@@ -0,0 +1,321 @@
+            #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+            dict_DACvsADC_Graphs[oh][vfat].GetYaxis().SetTitle(nameX)
+            if nominalDacValues[nameX][1][len(nominalDacValues[nameX][1])-1] == 'A':
+                dict_DACvsADC_Graphs[oh][vfat].GetXaxis().SetTitle(nameY + " (#muA)")
+            else:
+                dict_DACvsADC_Graphs[oh][vfat].GetXaxis().SetTitle(nameY + " (mV)")
+
+
+    outputFiles = {}         
+             
+    for oh in ohArray:
+        if scandate == 'noscandate':
+            outputFiles[oh] = r.TFile(elogPath+"/"+chamber_config[oh]+"/"+args.outfilename,'recreate')
+        else:    
+            outputFiles[oh] = r.TFile(dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate+"/"+args.outfilename,'recreate')
+
+    # Loop over events in the tree and fill plots
+    for event in dacScanFile.dacScanTree:
+        oh = event.link
+        vfat = event.vfatN
+
+        #the output of the calibration is mV
+        calibrated_ADC_value=calInfo[oh]['slope'][vfat]*event.dacValY+calInfo[oh]['intercept'][vfat]
+        calibrated_ADC_error=calInfo[oh]['slope'][vfat]*event.dacValY_Err
+
+        #Use Ohm's law to convert the currents to voltages. The VFAT3 team told us that a 20 kOhm resistor was used.
+        if nominalDacValues[nameX][1][len(nominalDacValues[nameX][1])-1] == "A":
+
+            #V (mV) = I (uA) R (kOhm)
+            #V (10^-3) = I (10^-6) R (10^3)
+            calibrated_ADC_value = calibrated_ADC_value/20.0
+            calibrated_ADC_error = calibrated_ADC_error/20.0
+
+            if nameX != 'CFG_IREF':
+                calibrated_ADC_value -= nominal_iref 
+            
+        #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+        #the nameX variable is the DAC register that is scanned, and we want to determine its nominal value
+        if args.assignXErrors:
+            dict_DACvsADC_Graphs[oh][vfat].SetPoint(dict_DACvsADC_Graphs[oh][vfat].GetN(),calibrated_ADC_value,event.dacValX)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPoint(dict_RawADCvsDAC_Graphs[oh][vfat].GetN(),event.dacValX,event.dacValY)
+            dict_DACvsADC_Graphs[oh][vfat].SetPointError(dict_DACvsADC_Graphs[oh][vfat].GetN()-1,0,event.dacValX_Err)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPointError(dict_RawADCvsDAC_Graphs[oh][vfat].GetN()-1,event.dacValX_Err,0)
+        else:
+            dict_DACvsADC_Graphs[oh][vfat].SetPoint(dict_DACvsADC_Graphs[oh][vfat].GetN(),calibrated_ADC_value,event.dacValX)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPoint(dict_RawADCvsDAC_Graphs[oh][vfat].GetN(),event.dacValX,event.dacValY)
+            dict_DACvsADC_Graphs[oh][vfat].SetPointError(dict_DACvsADC_Graphs[oh][vfat].GetN()-1,0,0)
+            dict_RawADCvsDAC_Graphs[oh][vfat].SetPointError(dict_RawADCvsDAC_Graphs[oh][vfat].GetN()-1,0,0)
```

Comment:
lines 229-238, you should always be applying `calibrated_ADC_error` here.  Why is this not applied?

Reply:
please see: https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/158#issuecomment-430004251

Reply:
Make a dummy plot if necessary for fitting, the errors *have physical meaning* and should be shown.

Reply:
Additionally if the user applies a `--stepSize` argument and chooses something other than 1 (e.g. 10, or 25) the number of points will drastically decrease.

Reply:
OK, so the procedure will be:

1) try to fit with the dacValY errors
2) check if the fit converges
3) if not, fit without the dacValY errors

In all cases, the output plots will show the dacValY errors.

Reply:
Sounds good.

You can also use the fit option “M” to have the minimized in Minuit try “more” to look for the solution.

Reply:
done